### DISCUSSION TITLE
How to add html in a specific place

### DISCUSSION THREAD

**Discussion by TheSeagullDev** [ScratchAddons/ScratchAddons#329] [Help]
Sorry if I'm missing something, but how do I add an html element to a specific part of a webpage? I'm trying to add a link to the scratch homepage.

**Comment by jeffalo** [comment]
use a userscript which is javascript and then commit [dom manipulation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)

**Comment by TheSeagullDev** [comment]
> use a userscript which is javascript and then commit [dom manipulation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)

Thanks, turns out scratch already has an element where I'm trying to add it, but it's empty. I googled how to change it, but it seems like the element needs an id, which this element doesn't have. Any suggestions?

**Comment by mxmou** [comment]
> > use a userscript which is javascript and then commit [dom manipulation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)
> 
> Thanks, turns out scratch already has an element where I'm trying to add it, but it's empty. I googled how to change it, but it seems like the element needs an id, which this element doesn't have. Any suggestions?

`document.querySelector()` with a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) as the argument.

**Comment by jeffalo** [comment]
> > > use a userscript which is javascript and then commit [dom manipulation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)
> > 
> > 
> > Thanks, turns out scratch already has an element where I'm trying to add it, but it's empty. I googled how to change it, but it seems like the element needs an id, which this element doesn't have. Any suggestions?
> 
> `document.querySelector()` with a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) as the argument.

this

**Comment by GrahamSH-LLK** [comment]
What do you mean by "this"?

**Comment by TheSeagullDev** [comment]
> > > use a userscript which is javascript and then commit [dom manipulation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)
> > 
> > 
> > Thanks, turns out scratch already has an element where I'm trying to add it, but it's empty. I googled how to change it, but it seems like the element needs an id, which this element doesn't have. Any suggestions?
> 
> `document.querySelector()` with a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) as the argument.

Is it possible to use multiple selectors, say only a `<p>` in a given class?

**Comment by mxmou** [comment]
> Is it possible to use multiple selectors, say only a `<p>` in a given class?

Yes, that would be `document.querySelector("p.class")`.

**Comment by TheSeagullDev** [comment]
Thank you
